File: knowledge/src/main/java/com/ayvytr/knowledge/contract/GankHistoryContentContract.java
package com.ayvytr.knowledge.contract;

import com.ayvytr.commonlibrary.bean.GankHistoryContent;
import com.ayvytr.mvpbase.IModel;
import com.ayvytr.mvpbase.IView;

import io.reactivex.Observable;

/**
 * @author admin
 */
public class GankHistoryContentContract {
    public interface Model extends IModel {
        Observable<GankHistoryContent> getGankByDate(String year, String month, String dayOfMonth);
    }

    public interface View extends IView {
        void showGankByDate(GankHistoryContent gankHistoryContent);
    }
}

File: knowledge/src/main/java/com/ayvytr/knowledge/contract/GankHistoryContentContract.kt
package com.ayvytr.knowledge.contract

import com.ayvytr.commonlibrary.bean.GankHistoryContent
import com.ayvytr.mvp.IModel
import com.ayvytr.mvp.IView

import io.reactivex.Observable

/**
 * @author admin
 */
class GankHistoryContentContract {
    interface Model : IModel {
        fun getGankByDate(year: String, month: String, dayOfMonth: String): Observable<GankHistoryContent>
    }

    interface View : IView {
        fun showGankByDate(gankHistoryContent: GankHistoryContent)
    }
}
